src/PrestaShopBundle/Resources/config/services/core/import.yml

```diff
@@ -27,3 +27,78 @@ services:
         class: 'PrestaShop\PrestaShop\Core\Import\Sample\SampleFileProvider'
         arguments:
             - '@prestashop.adapter.legacy.configuration'
+
+    prestashop.core.import.csv_file_reader:
+        class: 'PrestaShop\PrestaShop\Core\Import\File\CsvFileReader'
+        arguments:
+            - '@=service("prestashop.core.import.normalizer.csv_value_separator").normalize(service("session").get("separator"))'
```

Comment:
What if session is unavailable ? It looks like too many calls here for a service constructor. I think we need a dedicated [Service Factory](https://symfony.com/doc/current/service_container/factories.html).

Reply:
let's leave this for the next PR

Reply:
All right